Move Windows Mono to Staging (Closes: #49569)

directhex opened the pull request.

Commit:
Move Windows Mono to Staging (Closes: #49569)

stephentoub commented on eng/pipelines/runtime-staging.yml:
```diff
@@ -95,6 +95,48 @@ jobs:
+    runtimeFlavor: mono
+    platforms:
+    - Windows_x64
+    - Windows_x86
```
Does staging need both x64 and x86?

Commit:
no scenarios for windows

Commit:
Move x86 back how it was

Commit:
Move x64 runtime builds back to main, only library test run in staging

directhex commented:
It's not setting the target helix queues correctly, and danged if I can figure out why 😬 

I hate the black box that is AzDO YAML

Commit:
Windows/helix is where testGroup vs testScope really matters

Commit:
REVERT ME - test whether this variable is getting passed at all

Commit:
Revert "REVERT ME - test whether this variable is getting passed at all"

This reverts commit 3efbafae296c4ddc4fe875350fdd54fe2493326f.

Commit:
Use debug helix queue names

Commit:
More debug nonsense

Commit:
More variables-as-queue hackery

Commit:
Reenable windows non-staging so I can directly compare variables

Commit:
Use negation to avoid null in yaml

The opposite of `eq(parameters.jobParameters.isFullMatrix, true)`
is NOT `ne(parameters.jobParameters.isFullMatrix, false)`
because if parameters.jobParameters.isFullMatrix is null, an if
block comparing false _or_ true will get a null result.

As a result, running a helix job without explicitly setting a value
for isFullMatrix would fail on Windows, as Windows is the only
platform which does not specify any queues at all on innerloop
Mono jobs unless isFullMatrix is either false or true (but not null)

Commit:
Merge remote-tracking branch 'origin/main' into move-the-thing-to-staging

Commit:
Revert "Reenable windows non-staging so I can directly compare variables"

This reverts commit 32998940a13a7645434df6c216f7ef6bf9a72098.
